src/main/java/codeu/controller/ActivityFeedServlet.java

```diff
@@ -1,16 +1,203 @@
+    return new Activity(creationTime,output);
+  }
+  /**
+   * Formats Message data into readable string.
+   */
+  private Activity formatMessage(Message message) {
+    User author = userStore.getUserWithId(message.getAuthorId());
+    Conversation conversation = conversationStore.getConversationWithId(message.getConversationId());
+    String creationTime = formatTime(message.getCreationTime());
+    System.out.println(author);
+    System.out.println(conversation);
+    System.out.println(message);
+    String output = author.getName() + " sent in " + conversation.getTitle() + ": " + message.getContent();
+    return new Activity(creationTime,output);
+  }
+
+  /**
+   * Gets all activity on app, iterates through to sort by timestamp and prepares printed statements.
+   */
+  List<Activity> getActivities() {
+    List<Activity> activities = new LinkedList<>();
+    LinkedList<Conversation> conversations = new LinkedList<>(conversationStore.getAllConversations());
+    System.out.println(conversationStore);
+    LinkedList<User> users = new LinkedList<>(userStore.getAllUsers());
+    LinkedList<Message> messages = new LinkedList<>(messageStore.getAllMessages());
+
+    System.out.print("\n");
+    System.out.print("Length of conversations: ");
+    System.out.print(conversations.size());
+    System.out.print("\n");
+    System.out.print("Length of messages: ");
+    System.out.print(messages.size());
+    System.out.print("\n");
+    System.out.print("Length of users: ");
+    System.out.print(users.size());
+    System.out.print("\n");
+
+    while (!conversations.isEmpty() || !users.isEmpty() || !messages.isEmpty()) {
+      // Iterates through all new conversations, users and messages in the lists, compares the creation times of
+      //  each then inserts objects chronologically into list of activities.
+      ActivityType nextActivity = null;
+      Instant nextActivityTime = null;
+
+      if (!conversations.isEmpty()) {
+        nextActivity = ActivityType.ConversationCreated;
+        nextActivityTime = conversations.peek().getCreationTime();
+      }
+      if (!users.isEmpty()) {
+        if (nextActivityTime == null) {
+          nextActivity = ActivityType.UserJoined;
+          nextActivityTime = users.peek().getCreationTime();
+        }
+        else if (users.peek().getCreationTime().compareTo(nextActivityTime) < 0) {
+          nextActivity = ActivityType.UserJoined;
+          nextActivityTime = users.peek().getCreationTime();
+        }
+      }
+      if (!messages.isEmpty()) {
+        if (nextActivityTime == null) {
+          nextActivity = ActivityType.MessageSent;
+          nextActivityTime = messages.peek().getCreationTime();
+        }
+        if (messages.peek().getCreationTime().compareTo(nextActivityTime) < 0) {
+          nextActivity = ActivityType.MessageSent;
+          nextActivityTime = messages.peek().getCreationTime();
+        }
+      }
+
+      if (nextActivity != null) {
```

Comment:
Are there cases when this would be null? I expect not, and I might raise a fatal exception of some sort if it were, because that means something really bad happened